src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService_UpdateSource.cs

```diff
@@ -40,41 +40,59 @@ internal partial class DiagnosticAnalyzerService : IDiagnosticUpdateSource
             }
         }
 
-        internal void RaiseDiagnosticsUpdated(DiagnosticsUpdatedArgs args)
+        internal void RaiseDiagnosticsUpdated(ImmutableArray<DiagnosticsUpdatedArgs> args)
         {
             // all diagnostics events are serialized.
-            var ev = _eventMap.GetEventHandlers<EventHandler<DiagnosticsUpdatedArgs>>(DiagnosticsUpdatedEventName);
+            var ev = _eventMap.GetEventHandlers<EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>>>(DiagnosticsUpdatedEventName);
```

Comment:
>var ev = _eventMap.GetEventHandlers<EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>>>(DiagnosticsUpdatedEventName); [](http://example.com/codeflow?start=12&length=119)

check size before proceeding (or make sure all callers do before calling this)?

Reply:
⏱️ Currently all callers of this method check their count before calling this method, but I will relocate the check into this method for consistency with other events on the same paths.

Reply:
➡️ Now updated